core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/LazyAzureHttpClient.java

```diff
@@ -0,0 +1,92 @@
+package com.microsoft.applicationinsights.internal.channel.common;
+
+import com.azure.core.http.HttpClient;
+import com.azure.core.http.HttpRequest;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.http.ProxyOptions;
+import com.azure.core.http.netty.NettyAsyncHttpClientBuilder;
+import com.azure.core.util.Context;
+import com.microsoft.applicationinsights.customExceptions.FriendlyException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import reactor.core.publisher.Mono;
+import reactor.netty.resources.ConnectionProvider;
+
+import javax.annotation.concurrent.GuardedBy;
+import java.net.InetSocketAddress;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+public class LazyAzureHttpClient implements HttpClient {
+    private static final Logger logger = LoggerFactory.getLogger(LazyAzureHttpClient.class);
+    private static final HttpClient INSTANCE = new LazyAzureHttpClient();
+    private static final int DEFAULT_MAX_TOTAL_CONNECTIONS = 200;
+
+    public static volatile CountDownLatch safeToInitLatch;
+    public static volatile String proxyHost;
+    public static volatile Integer proxyPortNumber;
+    public static HttpClient getInstance() {
+        return INSTANCE;
+    }
+    private final Object lock = new Object();
+    @GuardedBy("lock")
+    private RuntimeException initException;
+    @GuardedBy("lock")
+    private HttpClient delegate;
+    private HttpClient getDelegate() {
+        synchronized (lock) {
+            if (delegate != null) {
+                return delegate;
+            }
+            if (initException != null) {
+                throw initException;
+            }
+            try {
+                delegate = init();
+            } catch (RuntimeException e) {
+                initException = e;
```

Comment:
i am curious to know why not throw here directly and wait for second called of this method then throw at line 42?